### ISSUE TITLE
setting "verify: true" is causing alexa app to timeout from echo devices

### ISSUE THREAD

**Issue by cpup22** [alexa-js/alexa-app-server#52]
When I set the flag for verify to be true it works when I try to check my endpoint from a non-echo device.  It properly returns a 401 error message:

```
{
  "status": "failure",
  "reason": "The signaturecertchainurl HTTP request header is invalid!"
}
```

However, when I try to test the app from the Amazon Alexa dev website or any echo devices, it throws an error that the requested skill has taken too long to respond.

Switching the flag back to false or removing completely and it goes back to working from echo devices but obviously it is going to fail certification.

**Comment by mreinstein** [comment]
I don't think middlewares are being handled correctly. There seems to be a lot of conflicting bodyparser related activity in the code:

https://github.com/alexa-js/alexa-app-server/blob/master/index.js#L166

this line declares a body parser *before* pulling in the raw body (https://github.com/alexa-js/alexa-app-server/blob/master/index.js#L171)

there is also installation of the `alexa-verifier-middleware` on line 85, which overlaps with line 171:

https://github.com/alexa-js/alexa-app-server/blob/master/index.js#L85

This is kind of a mess. I suspect this is why @cpup22 is having problems. Seems also possibly related:

https://github.com/alexa-js/alexa-app-server/issues/21

**Comment by tejashah88** [comment]
#35 might also be related?

**Comment by tejashah88** [comment]
Another thing to note, for some reason, the `alexa-verifier-middleware` is loaded **after** the url and json body parsers are loaded.

**Comment by mreinstein** [comment]
@tejashah88 are you available to chat on google or skype or slack? this will be too hard over github messaging.

**Comment by tejashah88** [comment]
Yes, I'm available to chat. I can do either slack or google.

**Comment by mreinstein** [comment]
I just pinged you on gchat

**Comment by dblock** [comment]
@cpup22 I have am incomplete fix in https://github.com/alexa-js/alexa-app-server/pull/64, and I have been using a test project from https://github.com/dblock/alexa-app-server-hello-world running on Heroku for testing this. If you want to see quickly if it solves your problem, you can try

```
  "dependencies": {
    "alexa-app-server": "dblock/alexa-app-server#a-little-closer-56"
  }
```

**Comment by cpup22** [comment]
@dblock I updated and did an npm install and see the following...

```
lisaseacat@meanstack-bno-vm:/opt/bitnami/apps/myapp$ npm install
blindnightout@1.0.0 /opt/bitnami/apps/myapp
└─┬ alexa-app-server@2.3.1  (git://github.com/dblock/alexa-app-server.git#fb76557626c72ee23bd2be8a1a5ee226ac28604b)
  ├── alexa-app@2.4.1  (git://github.com/dblock/alexa-app.git#5fa10102b03b4a67448846f2f3d4c1ca0d104a23)
  ├── alexa-verifier-middleware@0.1.9 
  └── lodash.defaults@4.2.0 

npm WARN blindnightout@1.0.0 No repository field.
lisaseacat@meanstack-bno-vm:/opt/bitnami/apps/myapp$ npm start

> blindnightout@1.0.0 start /opt/bitnami/apps/myapp
> node app.js

serving static content from: public
loading server-side modules from: server
   loaded /opt/bitnami/apps/myapp/server/login.js
loading apps from: apps
/opt/bitnami/apps/blindnightout/apps/helloworld/node_modules/alexa-app/index.js:491
    express.post(endpoint, function(req, res) {
            ^

TypeError: express.post is not a function
    at Object.express (/opt/bitnami/apps/myapp/apps/helloworld/node_modules/alexa-app/index.js:491:13)
```

Looks like the error is coming form alexa-app/index.js  Maybe I didnt' get the changes?? It looks based on the console log that it pulled from your github repo.

**Comment by dblock** [comment]
Doesn't look like it's using the right alexa-app code, try adding `alexa-app: "dblock/alexa-app#test-app-schema-utterances"` explicitly to package.json? If you ook at https://github.com/dblock/alexa-app/blob/test-app-schema-utterances/index.js#L491 which it's complaining about it doesn't have this code, but the shipped 2.4.0 version in https://github.com/alexa-js/alexa-app/blob/v2.4.0/index.js#L491 does.

**Comment by dblock** [comment]
Also, the good news is that #64 has an otherwise failing test for this issue that reproduces the same hang, so once we fix this we're not going to reintroduce a regression in future versions.
